lib/routes/account.js

```diff
@@ -1257,49 +1257,116 @@ module.exports = function (
         var authPW = Buffer(request.payload.authPW, 'hex')
         var authSalt = crypto.randomBytes(32)
         var password = new Password(authPW, authSalt, config.verifierVersion)
-        return password.verifyHash()
-          .then(
-            function (verifyHash) {
-              return db.resetAccount(
-                accountResetToken,
-                {
-                  authSalt: authSalt,
-                  verifyHash: verifyHash,
-                  wrapWrapKb: crypto.randomBytes(32),
-                  verifierVersion: password.version
-                }
-              )
-            }
-          )
-          .then(
-            function () {
-              // Notify all devices that the account has changed.
-              push.notifyUpdate(accountResetToken.uid, 'passwordReset')
+        var account, sessionToken, keyFetchToken, verifyHash, wrapKb
 
-              return db.account(accountResetToken.uid)
-            }
-          )
-          .then(
-            function (account) {
-              log.activityEvent('account.reset', request, {
-                uid: account.uid.toString('hex')
+        return verifyAndResetAccount()
+          .then(createSessionToken)
```

Comment:
I was thinking about the mechanism to indicate "provision me a sessionToken". If we go with `sessionToken: true` as part of the POST body, we'll need a new intermediate auth-server deployment to accept that field w/o blowing up, to facilitate rollback if necessary.